packages/ui/src/components/QueryBuilder/QueryPills/BooleanQueryPill.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import StackLayout from '../../../layout/StackLayout';
+import { ISyntheticSqon, IValueFilter, TSqonGroupOp } from '../../../data/sqon/types';
+import { isBooleanOperator, isReference } from '../../../data/sqon/utils';
+import ReferenceQueryPill from './ReferenceQueryPill';
+import FieldQueryPill from './FieldQueryPill';
+import Combiner from '../Combiner';
+import { IDictionary, TCallbackRemoveAction, TCallbackRemoveReferenceAction } from '../types';
+
+interface IBooleanQueryPillProps {
+    parentQueryId: string;
+    query: ISyntheticSqon | Record<string, never>;
+    isActive: boolean;
+    showLabels?: boolean;
+    dictionary: IDictionary;
+    onRemoveFacet: TCallbackRemoveAction;
+    onRemoveReference: TCallbackRemoveReferenceAction;
+    onCombineChange?: (id: string, combinator: TSqonGroupOp) => void;
+    getColorForReference?: (refIndex: number) => string;
+}
+
+const BooleanQueryPill: React.FC<IBooleanQueryPillProps> = (props) => {
+    const {
+        parentQueryId,
+        query,
+        isActive,
+        showLabels,
+        dictionary,
+        onRemoveFacet,
+        onRemoveReference,
+        onCombineChange = (f) => f,
+        getColorForReference = () => '',
+    } = props;
+    const { op, content } = query;
```

Comment:
please move desctructuring inside params and remove return statement.

Reply:
I need the props object for:
` <BooleanQueryPill {...props} query={f} />`